pkg/certificate/providers/vault/certificate_manager.go

```diff
@@ -30,180 +25,65 @@ const (
-		return cert, nil
-	}
-
-	cert, err := cm.issue(cn, validityPeriod)
-	if err != nil {
-		return cert, err
-	}
-
-	cm.cache.Store(cn, cert)
-
-	log.Trace().Msgf("Issued new certificate with SerialNumber=%s took %+v", cert.GetSerialNumber(), time.Since(start))
-
-	return cert, nil
-}
-
-// ReleaseCertificate is called when a cert will no longer be needed and should be removed from the system.
-func (cm *CertManager) ReleaseCertificate(cn certificate.CommonName) {
-	// TODO(draychev): implement Hashicorp Vault delete-cert API here: https://github.com/openservicemesh/osm/issues/2068
-	cm.deleteFromCache(cn)
-}
-
-// ListCertificates lists all certificates issued
-func (cm *CertManager) ListCertificates() ([]*certificate.Certificate, error) {
-	var certs []*certificate.Certificate
-	cm.cache.Range(func(cnInterface interface{}, certInterface interface{}) bool {
-		certs = append(certs, certInterface.(*certificate.Certificate))
-		return true // continue the iteration
-	})
-	return certs, nil
-}
-
-// GetCertificate returns a certificate given its Common Name (CN)
-func (cm *CertManager) GetCertificate(cn certificate.CommonName) (*certificate.Certificate, error) {
-	if cert := cm.getFromCache(cn); cert != nil {
-		return cert, nil
-	}
-	return nil, errCertNotFound
-}
-
 // GetRootCertificate returns the root certificate.
 func (cm *CertManager) GetRootCertificate() (*certificate.Certificate, error) {
-	return cm.ca, nil
-}
-
-// RotateCertificate implements certificate.Manager and rotates an existing certificate.
-func (cm *CertManager) RotateCertificate(cn certificate.CommonName) (*certificate.Certificate, error) {
-	start := time.Now()
-
-	oldCert, ok := cm.cache.Load(cn)
-	if !ok {
-		return nil, errors.Errorf("Old certificate does not exist for CN=%s", cn)
-	}
-
-	// We want the validity duration of the CertManager to remain static during the lifetime
-	// of the CertManager. This tests to see if this value is set, and if it isn't then it
-	// should make the infrequent call to configuration to get this value and cache it for
-	// future certificate operations.
-	if cm.serviceCertValidityDuration == 0 {
-		cm.serviceCertValidityDuration = cm.cfg.GetServiceCertValidityPeriod()
-	}
-	newCert, err := cm.issue(cn, cm.serviceCertValidityDuration)
+	// Create a temp certificate to determine the public part of the issuing CA
+	cert, err := cm.IssueCertificate("localhost", decade)
 	if err != nil {
 		return nil, err
 	}
 
-	cm.cache.Store(cn, newCert)
-
-	cm.msgBroker.GetCertPubSub().Pub(events.PubSubMessage{
-		Kind:   announcements.CertificateRotated,
-		NewObj: newCert,
-		OldObj: oldCert.(*certificate.Certificate),
-	}, announcements.CertificateRotated.String())
-
-	log.Debug().Msgf("Rotated certificate (old SerialNumber=%s) with new SerialNumber=%s took %+v", oldCert.(*certificate.Certificate).GetSerialNumber(), newCert.SerialNumber, time.Since(start))
-
-	return newCert, nil
+	//TODO(2068): implement a delete cert
```

Comment:
https://github.com/openservicemesh/osm/issues/2068 is closed. Does the TODO still apply?